### PR TITLE
mypy_primer: truncate per-project output

### LINKED ISSUE #14059: Truncating mypy_primer output per project
If there is a lot of mypy_primer output from a single project, the mypy primer is truncated so that output from some projects may not be visible at all. It could be better if we'd separately truncate the output of each project. We could still have a global output limit that is, for example, 3x the per-project output limit, to avoid huge output if a change breaks everything.

### PR DESCRIPTION
**Pull request by ikonst**
Closes #14059.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
The last commit didn't trigger mypy_primer at all, due to the fact that there's now a merge conflict :)

**Comment by ikonst** [comment]
So it didn't trigger the primer run but triggered the merging?

**Comment by ikonst** [comment]
OK, conflict was in tests, resolved the conflict in some random way.

**Comment by hauntsaninja** [comment]
IIRC you're not going to be able to test this in a PR to mypy, unfortunately. The comment workflow requires additional perms, so I believe it doesn't actually run the code in this PR. You could get it to run on your fork though!

**Comment by ikonst** [comment]
@hauntsaninja Thanks for the tip, makes perfect sense. I'm only starting to grok the security model of Actions.

**Comment by ikonst** [comment]
@hauntsaninja tested [here](https://github.com/ikonst/mypy/pull/2#issuecomment-1314541169)

**Comment by ikonst** [comment]
@hauntsaninja 👋

File: .github/workflows/mypy_primer_comment.yml [hunk 1]
```diff
@@ -48,15 +48,29 @@ jobs:
...
-            if (data.length > 30000) {
-              let truncated_data = data.substring(0, 30000)
-              let lines_truncated = data.split('\n').length - truncated_data.split('\n').length
-              data = truncated_data + `\n\n... (truncated ${lines_truncated} lines) ...\n`
+
+            function truncateIfNeeded(original, maxLength) {
+              let truncated = original.substring(0, maxLength)
+              if (original.length === truncated.length) {
+                return original
+              }
```
Comment (hauntsaninja) [inline .github/workflows/mypy_primer_comment.yml:61]: nit:
```suggestion
              if (original.length <= maxLength) {
                return original
              }
              let truncated = original.substring(0, maxLength)
```

**Review by hauntsaninja — APPROVED** [review]
Thanks, looks good!

**Pull request merged**